Microsoft.Toolkit.Uwp.UI/Extensions/NavigationView/NavigationViewExtensions.cs

```diff
@@ -0,0 +1,181 @@
+        {
+            var navview = d as Windows.UI.Xaml.Controls.NavigationView;
+
+            if (navview == null)
+            {
+                return;
+            }
+
+            navview.ItemInvoked -= Navview_ItemInvoked;
+
+            if (e.NewValue as bool? == true)
+            {
+                // Listen for clicks on navigation items
+                navview.ItemInvoked += Navview_ItemInvoked;
+            }
+            else
+            {
+                // Make sure we're visible if we toggle this off.
+                var content = navview.FindDescendantByName("ContentGrid");
+
+                if (content != null)
+                {
+                    content.Visibility = Visibility.Visible;
+                }
+            }
+        }
+
+        private static void Navview_ItemInvoked(NavigationView sender, NavigationViewItemInvokedEventArgs args)
+        {
+            var content = sender.FindDescendantByName("ContentGrid");
+
+            if (content != null)
+            {
+                // If we click the item we already have selected, we want to collapse our content
+                if (sender.SelectedItem != null && args.InvokedItem.Equals((sender.SelectedItem as NavigationViewItem).Content))
+                {
+                    // We need to dispatch this so the underlying selection event from our invoke processes.
+                    // Otherwise, we just end up back where we started.  We don't care about waiting for this to finish.
+                    #pragma warning disable CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
+                    sender.Dispatcher.RunIdleAsync((e) =>
+                    {
+                         sender.SelectedItem = null;
+                    });
+                    #pragma warning restore CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
+
+                    content.Visibility = Visibility.Collapsed;
+                }
+                else
+                {
+                    content.Visibility = Visibility.Visible;
+                }
+            }
+        }
+
+        private static void OnSelectedIndexChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var navview = d as Windows.UI.Xaml.Controls.NavigationView;
+
+            if (navview == null)
+            {
+                return;
+            }
+
+            navview.Loaded -= Navview_Loaded;
+            Navview_Loaded(d, null); // For changes
+            navview.Loaded += Navview_Loaded;
+
+            navview.SelectionChanged -= Obj_SelectionChanged;
+            navview.SelectionChanged += Obj_SelectionChanged;
+        }
+
+        private static void Navview_Loaded(object sender, RoutedEventArgs e)
+        {
+            if (sender != null && sender is NavigationView navview)
```

Comment:
Sender would never be null and will always be NavigationView a Cast would be more efficient :)